smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/ConfiguredChannelFactory.java

```diff
@@ -142,7 +144,13 @@ private static String getConnectorAttribute(Config config) {
         IncomingConnectorFactory mySourceFactory = incomingConnectorFactories.select(ConnectorLiteral.of(connector))
                 .stream().findFirst().orElseThrow(() -> new IllegalArgumentException("Unknown connector for " + name + "."));
 
-        return mySourceFactory.getPublisherBuilder(config);
+        PublisherBuilder<? extends Message> publisher = mySourceFactory.getPublisherBuilder(config);
+
+        for (PublisherDecorator decorator : CDI.current().select(PublisherDecorator.class)) {
+            publisher = decorator.decoratePublisher(publisher, name);
```

Comment:
What about the order? Should we introduce some kind of `ordinal`?

Reply:
We could do. At the moment there's only one. Would you like me to add an ordinal now or wait until we have another decorator use case?

Reply:
We can wait I believe.